components/timezone/src/types.rs

```diff
@@ -18,6 +18,11 @@ impl Default for GmtOffset {
     }
 }
 
+fn try_get_time_component(chars: &[u8; 2]) -> Option<i32> {
```

Comment:
nit: passing `[u8; 2]` by reference is inefficient, because a pointer is bigger than the type itself (and the type is `Copy`). However this will almost certainly inlined, so it doesn't matter.

Reply:
Good point. I will check with a compiler to see what assembly it generates. clippy complains about the reference to slice without length. But not when the pointer is larger than the actual data size and length. I am wondering if the compiler or stack keeps data aligned with the target arch register size.

Reply:
In general in Rust folks tend to not care about by-value vs by-reference performance, for Copy types everything except large ones are typically passed by value. It tends to be the faster and also it rarely matters due to inlining)

(so here we should pass by value)

Reply:
[No reference](https://rust.godbolt.org/z/arKfKr5oM) (L182-193) comparing to [with reference](https://rust.godbolt.org/z/forvY9bf4). Exactly the same! So `[u8; 2]` is better.